tests/prefetch_related/tests.py

```diff
@@ -256,6 +259,81 @@ def test_m2m_then_reverse_one_to_one_object_ids(self):
         self.assertWhereContains(sql, self.author1.id)
 
 
+@mock.patch('django.db.connection.features.max_query_params', 50)
+class LargePrefetchSetTests(TestCase):
+    """
+    Prefetch queries using large "WHERE id IN (…)" clauses are performed in
+    batches on databases with a limited number of variables per query, such as
+    SQLite. To improve test performance, a smaller value of max_query_params
+    is mocked.
+    """
+    test_size = 101
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.book1 = Book.objects.create(title='Poems')
+        cls.book2 = Book.objects.create(title='Poems II')
+        Author.objects.bulk_create((
+            Author(name='Charlotte {}'.format(i), first_book=cls.book1)
+            for i in range(cls.test_size)
+        ))
+        cls.author_first = Author.objects.first()
+        cls.author_last = Author.objects.last()
+        cls.reader = Reader.objects.create(name='Amy')
+        # Add related objects to the first and the last authors in the list to
+        # ensure that the result sets are properly merged after the bulk
+        # processing.
+        cls.author_first.books.add(cls.book1)
+        cls.author_last.books.add(cls.book2)
+        cls.reader.books_read.add(cls.book1)
+        cls.reader.books_read.add(cls.book2)
+
+    @property
+    def num_batches(self):
+        if connection.features.max_query_params:
+            # 25 is the hardcoded safety margin to cope with additional
+            # variables in the query.
+            batch_size = connection.features.max_query_params - 25
```

Comment:
I wonder if 25 should be defined in the [SQL constants module](https://github.com/django/django/blob/master/django/db/models/sql/constants.py)?

I am afraid changing 25 in the code might not be changed here, so the test would silently become obsolete.